### ISSUE TITLE
How to disable error boundary overlay from babel/webpack config?

### ISSUE THREAD

**Issue by mohsinulhaq** [gaearon/react-hot-loader#1153]
We provide `babel/webpack` config to to other teams which have react-hot-loading enabled. And the users setup AppContainer/hot themselves. For them, it seems like a breaking change because going through docs, I find that `{errorBoundary: false}` prop to `AppContainer` or `hot` function is the only way to disable the overlay.

**Comment by theKashey** [comment]
https://github.com/gaearon/react-hot-loader#error-boundary-is-inside-every-component
Just provide some default value, to "disable" it.
```js
// to disable
setConfig({ ErrorOverlay: () => null })
```

**Comment by mohsinulhaq** [comment]
But this needs to happen on client side at runtime right?

**Comment by mohsinulhaq** [comment]
I want to disable it from webpack/babel config side

**Comment by theKashey** [comment]
It's not a part of webpack/babel plugin -as you said - it's _runtime_ React `ErrorBoundary` inside `AppContainer`, which is created inside `hot`, plus `hot` has HMR _error boundary_.

You may disable it via webpack config by adding one more file to the entry point, which will `setConfig`. Ie - 
```js
{
  entry: ['src/index.js', 'src/setup-hmr'], //(could be last, as long affect runtime, not startup time)
}
```

**Comment by mohsinulhaq** [comment]
Hi, your solution worked. But I see that nothing renders on componentDidCatch now. I am using redbox-react. Earlier redbox-react used to work on error. Now it seems RHL just turned off rendering anything on error. It just unmounts till the nearest error boundary.

**Comment by theKashey** [comment]
Could you please provide some pseudo code to reproduce?

**Comment by jeremym** [comment]
@theKashey -- I don't believe the issue @mohsinulhaq is having can be solved the way you suggest without a change to `AppContainer.dev.js`.  Meaning: the AppContainer class already has a `componentDidCatch` method on it (making it a react error boundary), so trying to add another Error Boundary in the chain will not catch an error that has already been caught.

https://github.com/gaearon/react-hot-loader/blob/f15b108cfac2aa6b8f98496b1e3103a4ad1c9e9c/src/AppContainer.dev.js#L39

**Comment by theKashey** [comment]
The change you are looking for is already in master and soon will be released.
In short - AppContainer will not catch errors if it's out of "Update window" - https://github.com/gaearon/react-hot-loader/blob/master/src/AppContainer.dev.js#L42

**Comment by theKashey** [comment]
Released in v4.6.4

**Comment by mohsinulhaq** [comment]
Thanks @theKashey ! Closing this issue.
